src/metaopt/worker/trial.py

```diff
@@ -0,0 +1,200 @@
+
+        :returns: a list of corresponding `Trial` objects.
+        """
+        trials = []
+        for entry in trial_entries:
+            trials.append(cls(**entry))
+        return trials
+
+    class Value(object):
+        """Container for a value object.
+
+        Attributes
+        ----------
+        name : str
+           A possible named for the quality that this is quantifying.
+        type : str
+           An identifier with semantic importance for **MetaOpt**. See
+           `Param.type` and `Result.type`.
+        value : str or numerical
+           value suggested for this dimension of the parameter space.
+
+        """
+
+        __slots__ = ('name', '_type', 'value')
+        allowed_types = ()
+
+        def __init__(self, **kwargs):
+            """See attributes of `Value` for possible argument for `kwargs`."""
+            for attrname in self.__slots__:
+                setattr(self, attrname, None)
+            for attrname, value in six.iteritems(kwargs):
+                setattr(self, attrname, value)
+
+        def __iter__(self):
+            """Needed to be able to convert `Value` to `dict` form."""
+            for attrname in self.__slots__:
+                if attrname == '_type':
+                    yield ('type', self._type)
+                else:
+                    yield (attrname, getattr(self, attrname))
+
+        def __str__(self):
+            """Represent partially with a string."""
+            ret = "{0}(name={1}, type={2}, value={3})".format(
+                type(self).__name__, repr(self.name),
+                repr(self._type), repr(self.value))
+            return ret
+
+        __repr__ = __str__
+
+        @property
+        def type(self):
+            """For meaning of property type, see `Value.type`."""
+            return self._type
+
+        @type.setter
+        def type(self, type_):
+            if type_ is not None and type_ not in self.allowed_types:
+                raise ValueError("Given type, {0}, not one of: {1}".format(
+                    type_, self.allowed_types))
+            self._type = type_
+
+    class Result(Value):
+        """Types for a `Result` can be either an evaluation of an 'objective'
+        function or of an 'constaint' expression.
+        """
+
+        allowed_types = ('objective', 'constraint')
+
+    class Param(Value):
+        """Types for a `Param` can be either an integer (discrete value),
+        floating precision numerical or a categorical expression (a string).
+        """
+
+        allowed_types = ('int', 'float', 'enum')
+
+    __slots__ = ('exp_name', 'user', '_id', '_status', 'worker',
+                 'submit_time', 'start_time', 'end_time', 'results', 'params')
+    allowed_stati = ('new', 'reserved', 'suspended', 'completed', 'broken')
+
+    def __init__(self, exp_name, user, **kwargs):
+        """See attributes of `Trial` for meaning and possible arguments for `kwargs`."""
+        for attrname in self.__slots__:
```

Comment:
[Documentation on __slots__](https://docs.python.org/3.6/reference/datamodel.html#slots)

I use them so that we prohibit accidental attribute usage, which will make future debugging harder. It's also a way to document a container like this that we do not want to accept anything else than what we designed to contain. It has small performance gains too.